src/stream/src/executor/source/source_backfill_executor.rs

```diff
@@ -442,23 +441,7 @@ impl<S: StateStore> SourceBackfillExecutorInner<S> {
             .state_store()
             .clone();
         let table_id = self.backfill_state_store.state_store().table_id().into();
-        static STATE_TABLE_INITIALIZED: Once = Once::new();
-        tokio::spawn(async move {
-            // This is for self.backfill_finished() to be safe.
-            // We wait for 1st epoch's curr, i.e., the 2nd epoch's prev.
-            let epoch = first_epoch.curr;
-            tracing::info!("waiting for epoch: {}", epoch);
-            state_store
-                .try_wait_epoch(
-                    HummockReadEpoch::Committed(epoch),
-                    TryWaitEpochOptions { table_id },
-                )
-                .await
-                .expect("failed to wait epoch");
-            STATE_TABLE_INITIALIZED.call_once(|| ());
-            tracing::info!("finished waiting for epoch: {}", epoch);
-        });
-
+        let mut state_table_initialized = false;
```

Comment:
Due to https://github.com/risingwavelabs/risingwave/issues/18312, the first barrier will be yield immediately, and the correctness doesn't hold any more.

Reply:
cc @wenym1 to take a look at the new solution.